libraries/AP_Airspeed/AP_Airspeed_SST_ND.cpp

```diff
@@ -0,0 +1,306 @@
+uint8_t sst_config_setting[2] = {0x0A, 0x07}; //bw limit set to 50Hz -> 155.35Hz, pressure range set to 0b010
+
+AP_Airspeed_SST_ND::AP_Airspeed_SST_ND(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+// probe for a sensor
+bool AP_Airspeed_SST_ND::probe(uint8_t bus, uint8_t address)
+{
+    _dev = hal.i2c_mgr->get_device(bus, address);
+    if (!_dev) {
+        return false;
+    }
+    WITH_SEMAPHORE(_dev->get_semaphore());
+
+    _dev->set_retries(20);
+    uint8_t reading[14]= {'\0'};
+    uint8_t model[8] = {'\0'};
+
+    if(!_dev->read(reading, 14)){
+        return false;
+    }else{
+        for (int i=0; i<14; i++){
+            if(i>5){
+                model[i-6]= reading[i];
+            }
+        }
+    }
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Found bus %u addr 0x%02x", _dev->bus_num(), _dev->get_bus_address());
+    return matchModel(model);
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_SST_ND::init()
+{
+    static const uint8_t addresses[] = { ND_I2C_ADDR1, ND_I2C_ADDR2 };
+    if (bus_is_confgured()) {
+        for (uint8_t addr : addresses) {
+            if (probe(get_bus(), addr)) {
+                goto found_sensor;
+            }
+        }
+    } else {
+        FOREACH_I2C_EXTERNAL(bus) {
+            for (uint8_t addr : addresses) {
+                if (probe(bus, addr)) {
+                    goto found_sensor;
+                }
+            }
+        }
+        FOREACH_I2C_INTERNAL(bus) {
+            for (uint8_t addr : addresses) {
+                if (probe(bus, addr)) {
+                    goto found_sensor;
+                }
+            }
+        }
+    }
+
+    //GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "SST_ND[%u]: no sensor found", get_instance());
+    return false;
+
+found_sensor:
+    _dev->set_device_type(uint8_t(DevType::SST_ND));
+    set_bus_id(_dev->get_bus_id());
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"SST_ND[%u]: Found bus %u addr 0x%02x", get_instance(), _dev->bus_num(), _dev->get_bus_address());
+
+    // send default configuration
+    WITH_SEMAPHORE(_dev->get_semaphore());
+    //_dev->transfer(config_setting, 2, nullptr,0);
+
+    switch(_dev_model){
+        case DevModel::SST_ND:
+            _available_ranges = 8;
+            _range_setting = 7;
+            _current_range_val = vn131cm_range[_range_setting];
+            break;
+        case DevModel::ND210:
+            _available_ranges = 7;
+            _range_setting = 3;
+            _current_range_val = nd210_range[_range_setting];
+            break;
+        case DevModel::ND005D:
+            _available_ranges = 6;
+            _range_setting = 3;
+            _current_range_val = nd005d_range[_range_setting];
+            break;
+        default:
+            //GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"ND not setup correctly");
+            return false;
```

Comment:
```suggestion
```

Should have no cause to have a `default:` clause here - should always know what to do for every enum entry.

Reply:
Removed `default:` and completed any missing definitions